telegram/ext/filters.py

```diff
@@ -2545,3 +2548,74 @@ def filter(self, message: Message) -> bool:
 
 VOICE = _Voice("filters.VOICE")
 """Messages that contain :attr:`telegram.Message.voice`."""
+
+
+class Mention(MessageFilter):
+    """Mention messages. If list of username or chat_id or `User` objects passed, it filters
+    messages to only allow those whose contains one of username  or chat_id or `User`.
+    Note:
+        Usernames only allowed with @ prefix. Example: "@username"
+        Chat id only allowed with positive numbers. Example: 123456
+    Examples:
+        A simple use case for passing username is to allow only messages that were sent with
+        mention username:
+        MessageHandler(filters.Mention("@username"), callback_method)
+        MessageHandler(filters.Mention(["@username",`123456`,`user`), callback_method)
+    Args:
+        mentions (Union[:obj:`str`,:obj:`str`,:obj:`User`] |
+        Collection[Union[:obj:`str`,:obj:`str`,:obj:`User`]]): Which messages to allow with
+        username or chat_id or `User` object.If one of them matched it will allow.
+        If not specified, will allow any text message with mention.
+    """
+
+    __slots__ = ("mention", "_mentions")
+
+    def __init__(self, mentions: SCT[Union[int, str, TGUser]]):
+        self._mentions = self._parse_mentions(mentions)
+        super().__init__(name=f"filters.Mention({mentions})")
+
+    @staticmethod
+    def _parse_mentions(
+        mentions: SCT[Union[int, str, TGUser]]
+    ) -> Iterable[Union[int, str, TGUser]]:
+        if isinstance(mentions, Iterable) and not isinstance(mentions, str):
+            return set(mentions)
+        return set(mentions)
+
+    @property
+    def mentions(self) -> FrozenSet[Union[int, str, TGUser]]:
+        return frozenset(self._mentions)
+
+    def check_mention(self, message: Message, mention: Union[int, str, TGUser]) -> bool:
```

Comment:
```suggestion
    @staticmethod
    def _check_mention(message: Message, mention: Union[int, str, TGUser]) -> bool:
```

No need to make the method part of documented behavior :)